### ISSUE TITLE
Beepedia Improvements

### ISSUE THREAD

**Issue by TemperedFool** [Team-Resourceful/ResourcefulBees#226]
**Is your feature request related to a problem? Please describe.**
Problem is a stretch, I think there's a few ways the Beepedia could be improved.

**Describe the solution you'd like**

- Allow the interface to expand and use more of the screen. I'm not sure if you'd want to tie this to GUI Scale somehow, but probably not. I'd likely include a scaling button in the interface, kind of like RS does. On a decently sized monitor the interface is hilariously small, and could display a lot more info. Screenshot of my personal setup at the end.
- Sorting within the Beepedia seems to use the bee Type rather than the display name from the lang file - this makes the sorting a bit weird. My current example is the Creeper and Coal bees, which I've renamed Beeper and Sooty Bee respectively; these sort within C rather than where you'd expect. If sorting is alphabetical by name, it should probably sort based on the name from the lang file in use. This seems to happen with honey too, and I would assume also traits.
- It would be very nice for there to be either sorting options or a filter to only display world-spawning or breadable bees. I imagine that the players, when they get started, would find an easy way to figure out "these are the bees I'm hunting" super useful.
- List the biome tags set for spawning for bees that have them. We have a lot of biomes, and the world-spawning bees have 3-5 JEI pages worth of biomes listed, but if they could see "Forest" or "Cold" they could generalize a bit easier (and also report biomes I've missed a bit easier).

**Describe alternatives you've considered**
Most of this currently CAN be done via either the Beepedia or JEI, this would just improve the experience.

**Additional context**

<img width="1908" alt="image" src="https://user-images.githubusercontent.com/13169307/115575474-97297900-a290-11eb-9e4b-03ef7ff79043.png">

**Comment by Vaerys-Dawn** [comment]
> Allow the interface to expand and use more of the screen. I'm not sure if you'd want to tie this to GUI Scale somehow, but probably not. I'd likely include a scaling button in the interface, kind of like RS does. On a decently sized monitor the interface is hilariously small, and could display a lot more info. Screenshot of my personal setup at the end.

The interface scales with ui scaling, there really isn't a good way to have the interface change size without MAJOR redesigns that aren't really worth the time or effort so this likely won't happen.

> Sorting within the Beepedia seems to use the bee Type rather than the display name from the lang file - this makes the sorting a bit weird. My current example is the Creeper and Coal bees, which I've renamed Beeper and Sooty Bee respectively; these sort within C rather than where you'd expect. If sorting is alphabetical by name, it should probably sort based on the name from the lang file in use. This seems to happen with honey too, and I would assume also traits.

I don't think this will change as each bee is stored in a map with their bee type as the key, in order to sort the list by the Lang text I would have to force the map to be converted into something else. It is possible but again it's hard to say if it's worth the time or effort

> It would be very nice for there to be either sorting options or a filter to only display world-spawning or breadable bees. I imagine that the players, when they get started, would find an easy way to figure out "these are the bees I'm hunting" super useful.

I'd love to add this too, but as of right now i'm unsure how to add search filters like this, expect to see something like this in the future.

> List the biome tags set for spawning for bees that have them. We have a lot of biomes, and the world-spawning bees have 3-5 JEI pages worth of biomes listed, but if they could see "Forest" or "Cold" they could generalize a bit easier (and also report biomes I've missed a bit easier).

This could be added, I'll have to figure out the best way to display it but it's definitely something worth looking into

**Comment by Epic428** [comment]
I agree with everything dawn has said. My only gripe with the 4th item is the fact that biome “tags” aren’t a thing in vanilla. So simply stating “forest” or “cold” seems great at first but there’s absolutely no way for a player to know what exactly falls in that category. Take the Oreo bee as an example. It uses “rare”. What is considered “rare” to the player? How would a player know what biomes belong to that “tag”? You can’t look it up on a wiki someplace Bc if you’re not using our custom tagging system then it’s all strictly handled by forge and automated based on certain data points about a biome. While having a massive list of biomes can be overwhelming when you pick something like “overworld”, it’s a lot better than having tags that a player can’t easily discern what’s actually inside of it. 

perhaps the better solution would be to identify a better way of displaying such a large list of biomes so that the list can be skimmed through quickly by a player.

**Comment by Epic428** [comment]
Also Lang files should be used primarily for the purpose of translations not so much altering creeper to beeper. I get it though Bc we do it too. But storing the data twice makes no sense. And constant lookups to the display name just to make searching a little better isn’t quite worth the effort. Hell even jei doesn’t search by display name Bc the same exact issue happens there as well. And we do bee types based on the name of the bee json. So ideally that should be where the name is set. Unfortunately you can’t just change the name of a json Bc of how everything is registered and doing so would break people’s worlds. So it’s better to have a standard set on a new mc version where people are forced to create new worlds.

**Comment by TemperedFool** [comment]
Yeah, the concern with breaking everything that already exists (I'm jumping into an already-implemented setup) means changing the actual bee types seems like it would break too much. We'll have a world-breaking update at some point, so maybe I'll take that time to do so - that will at least resolve the sorting issues.

If you mean it gets bigger/smaller with GUI Scaling, I see that (I can provide screenshots but I don't think you need them), but the amount of info displayed at any given scale is very small. I totally get it if it's too difficult to modify, but it's a shame - it would be very handy to have it fill up more of the screen and display more information.

**Comment by TemperedFool** [comment]
> Take the Oreo bee as an example. It uses “rare”. What is considered “rare” to the player?

Yeah, I definitely see that some of the tags (this is a very good example) don't mean too much, but I wouldn't eliminate the biome list, so that's always a fallback. For most of the bees I'm personally using tags like Forest and Cold, which although players could always get tricked a bit (I could have missed an appropriate biome in the dictionary files, or mis-applied one), it will give them a direction to head.
